docs/proposals/20200720-single-controller-multitenancy.md

```diff
@@ -0,0 +1,548 @@
+Cluster API operators and managing resources in the dev Subscription. Dascha can provision a new 
+cluster by creating Cluster API resources in the existing cluster, omitting the ProvisionerIdentity 
+field in the AzureCluster spec. The CAPZ operator will use the Azure credentials provided in its 
+deployment template.
+
+#### Story 3 - Software as a Service Provider
+
+ACME Industries is offering Kubernetes as a service to other organizations. ACME creates an AAD 
+Identity for each organization and each organization grants that Identity access to provision 
+infrastructure in one or multiple of their Azure Subscriptions. ACME Industries wants to minimise 
+the memory footprint of managing many clusters, and wants to move to having a single instance of 
+CAPZ to managed infrastructure across multiple organizations.
+
+## Requirements
+
+### Functional
+
+<a name="FR1">FR1.</a> CAPZ MUST support assuming an identity specified by an `AzureCluster.ProvisioningIdentity`.
+
+<a name="FR2">FR2.</a> CAPZ MUST support static credentials.
+
+<a name="FR3">FR3.</a> CAPZ MUST prevent privilege escalation allowing users to create clusters in Azure accounts they should
+  not be able to.
+
+<a name="FR4">FR4.</a> CAPZ SHOULD support credential refreshing modified principal data.
+
+<a name="FR5">FR5.</a> CAPZ SHOULD provide validation for principal data submitted by users.
+
+<a name="FR6">FR6.</a> CAPZ MUST support clusterctl move scenarios.
+
+### Non-Functional
+
+<a name="NFR1">NFR1.</a> Each instance of CAPZ SHOULD be able to support 200 clusters using role assumption.
+
+<a name="NFR2">NFR2.</a> CAPZ MUST call STS APIs only when necessary to prevent rate limiting.
+
+<a name="NFR3">NFR3.</a> Unit tests MUST exist for all credential provider code.
+
+<a name="NFR4">NFR4.</a> e2e tests SHOULD exist for all credential provider code.
+
+<a name="NFR5">NFR5.</a> Credential provider code COULD be audited by security engineers.
+
+### Implementation Details/Notes/Constraints
+
+The current implementation of CAPZ requests new instances of Azure services per cluster and 
+sub-cluster resources. The input for these services is a ClusterScope, which provides the identity
+Azure service will operate as.
+
+```go
+type ClusterScope struct {
+	logr.Logger
+	client      client.Client
+	patchHelper *patch.Helper
+
+	AzureClients
+	Cluster      *clusterv1.Cluster
+	AzureCluster *infrav1.AzureCluster
+}
+```
+The ClusterScope contains all of the infromation needed to make an authenticated request against an
+Azure cloud. The `Authorizer` currently contains the AAD identity information loaded from the CAPZ
+controller environment. The `Authorizer` is used to fetch and refresh tokes for all Azure clients.
+
+```go
+// AzureClients contains all the Azure clients used by the scopes.
+type AzureClients struct {
+	SubscriptionID             string
+	ResourceManagerEndpoint    string
+	ResourceManagerVMDNSSuffix string
+	Authorizer                 autorest.Authorizer
```

Comment:
this actually changes in #802

Reply:
I don't think this is a huge deal, I'd rather clean up our credentials anyway. I'd consider the changes to this in #802 mostly a workaround to avoid bigger changes